### DISCUSSION TITLE
TreeData with latest slickgrid version

### DISCUSSION THREAD

**Discussion by sapnildessai** [ghiscoding/Angular-Slickgrid#818] [General]
Hi @ghiscoding ,

Sorry to bother you again, but I am facing a strange issue. As soon as the Formatter in column definition is used it starts giving below error
![image](https://user-images.githubusercontent.com/17919638/127607018-8e58f082-bae1-43e8-b0d7-b225fccb60f1.png)

![image](https://user-images.githubusercontent.com/17919638/127606769-ef0929e7-6887-4896-896f-c2c64a2ab355.png)

If the formatter in column definition is removed everything works fine.

Do you have any idea what could go wrong?

Thanks

**Comment by sapnildessai** [comment]
This is also blocking me from using complexObject formatter :(

**Reply by ghiscoding** [reply]
just came back from vacations... 

I don't see enough info to help you (the demos are working fine too so...), you can troubleshoot yourself in Chrome by clicking on the line giving the error. I can't help without knowing the error from the formatter (which you didn't provide). At the end of the day, that is a Formatter and every Formatters can be reimplemented in your local code, you can take a look at the [Tree Formatter](https://github.com/ghiscoding/slickgrid-universal/blob/master/packages/common/src/formatters/treeFormatter.ts)

What is the `import` for that Formatter? Do you have other Formatters or only that one? Maybe look at the [demo](https://github.com/ghiscoding/angular-slickgrid-demos/blob/master/bootstrap4-demo-with-locales/src/app/examples/grid-tree-data-parent-child.component.ts)!?

**Reply by sapnildessai** [reply]
Hi @ghiscoding ,
yes. I had to copy complex formatter into my code and implement it in the way i want. This has resolved my problem for now. But i would debug it at some point of time why the default formatter is not working
Thanks
